RosettaImmutables/src/main/java/com/hubspot/rosetta/immutables/RosettaAwareWireSafeEnumSerializer.java

```diff
@@ -0,0 +1,68 @@
+package com.hubspot.rosetta.immutables;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.BeanProperty;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.ContextualSerializer;
+import com.hubspot.immutables.utils.WireSafeEnum;
+
+public class RosettaAwareWireSafeEnumSerializer extends JsonSerializer<WireSafeEnum<?>> implements ContextualSerializer {
+  private static final Map<Class<?>, JsonSerializer<WireSafeEnum<?>>> SERIALIZER_CACHE = new ConcurrentHashMap<>();
+
+  @Override
+  public JsonSerializer<?> createContextual(SerializerProvider prov, BeanProperty property) throws JsonMappingException {
+    JavaType contextualType = property.getType();
```

Comment:
Could this code be shared with `RosettaAwareWireSafeEnumDeserializer` somehow? Like maybe `Optional<Class<?>> getContextualClass` or whatever?